### DISCUSSION TITLE
Conan 2.0 Release and Conan Center support

### DISCUSSION THREAD

**Discussion by jcar87** [conan-io/conan-center-index#16196] [Announcements]
Conan 2.0 [is now available](https://blog.conan.io/2023/02/22/Conan-2.0.html).

Currently, Conan Center is home to recipes to almost 1500 libraries and tools. We estimate over 900 of the recipes in the repository have been migrated to be compatible with both Conan 1.X and Conan 2.0, thanks to the efforts of our community contributors.

On release day, we are focusing on ensuring that the recipes and packages for the most downloaded recipes (as per our access logs) are compatible and validated for Conan 2.0. Please see below for a full list of Conan 2.0-ready recipes. 

## List of validated recipes compatible with Conan 2.0:

Tested with packages available for the following configurations:

- Linux: `gcc` 11 (shared and static)
- macOS: `apple-clang` 13 (shared and static)
- Windows: `msvc` 192 (Visual Studio 2019) and `msvc` 193 (Visual Studio 2022) (shared and static)

**Validated recipes**: For an up-to-date list, please check this  ➡️ ➡️ ➡️  [list](https://github.com/conan-io/conan-center-index/blob/master/.c3i/conan_v2_ready_references.yml).

---

### Note on profiles and configurations
Conan 2.0 introduces a number of changes and improvements to settings in profiles:
* `compiler.cppstd` setting on all platforms: Conan 2.0 profile autodetection will fill out the cppstd setting. While this was available on Conan 1.X, it was not set by default. This improves our ability to model C++ standard compatibility when resolving a dependency graph.
* `msvc` compiler setting and valid versions: On Windows, the compiler setting is now called msvc and follows the version scheme of the C++ compiler rather than the version of the Visual Studio release. This more accurately reflects the compatibility, since the same version of Visual Studio can work with multiple toolchains (and therefore, multiple C++ compiler versions).
* `libstdc++11` autodetection: (libstdc++ with C++ ABI) is now autodetected where needed. This should now apply to most recent Linux distributions. For Conan 2.0 packages, configurations with the legacy ABI are no longer provided or tested.

### Building and consuming shared libraries
Conan 2.0 has an improved dependency graph that takes into account library types. For libraries that support being built as static and shared libraries, Conan Center will generate packages for both variants. When building the shared library variant, we will make sure that all dependencies are also shared libraries, to avoid any instances of symbols from static libraries being present in multiple places in the dependency graph. 
When consuming shared libraries, please ensure that this applies to the entire dependency graph:

```bash
conan install –require=zlib/1.2.13 –require=fmt/9.1.0 -o'*:shared=True'
```

We’re very excited for you to try Conan 2.0 and the new features it has to offer. If you have any issues, please reach out to us and report them:

- [Conan Center issue tracker](https://github.com/conan-io/conan-center-index/issues)
- [Conan issue tracker](https://github.com/conan-io/conan/issues)

**Comment by kkhoylou** [comment]
Will there be movement in validating the recipe for openssl/3.0.8? I believe it isn't validated currently

**Reply by Croydon** [reply]
There is some work done in https://github.com/conan-io/conan-center-index/pull/14426

After it's merged I plan to create a PR to adjust the rest

**Comment by leinadme** [comment]
I'm sure I'm writing in the wrong place, but I have an issue with Conan on Windows, and I can't seem to find a wiki or location where I can ask a question and have someone from the community help me.  Can someone please provide a link if it exists?  Sorry for the inconvenience.

**Comment by leinadme** [comment]
Never mind, I think I found it.

**Comment by hkk2023** [comment]
Conan Version : 2.0.16

- Trying to Create a package for graphviz version 2.49.3. One of the dependencies is glib and the glib version range is defined as ">=2.78.1 <3". Package graphviz also has dependency on harfbuzz version 8.3.0 and harfbuzz requires glib and is hardcoded to version 2.78.1.
- The latest glib version is 2.78.3

While creating packaging for graphviz, it takes latest glib version 2.78.3 but harfbuzz requires glib 2.78.1 which is resulting in a version conflict.

How can this conflict be resolved such that graphviz package can be created without any issues?

![image](https://github.com/conan-io/conan-center-index/assets/127183435/8e424799-5f48-486f-a4ee-29fdbb2440ff)
